lib/logstash/outputs/elasticsearch.rb

```diff
@@ -354,17 +356,13 @@ def receive(event)
   end # def receive
 
   public
-  # synchronize the @current_client.bulk call to avoid concurrency/thread safety issues with the
-  # # client libraries which might not be thread safe. the submit method can be called from both the
-  # # Stud::Buffer flush thread and from our own retry thread.
+  # The submit method can be called from both the
+  # Stud::Buffer flush thread and from our own retry thread.
   def submit(actions)
     es_actions = actions.map { |a, doc, event| [a, doc, event.to_hash] }
-    @submit_mutex.lock
-    begin
-      bulk_response = @current_client.bulk(es_actions)
-    ensure
-      @submit_mutex.unlock
-    end
+
+    bulk_response = @client.bulk(es_actions)
```

Comment:
In that case it IS a good idea for us to not do a final release till those thread safety patches make it into the ruby client. Good catch @talevy